### PR TITLE
dtj chapter 7

### PR DESCRIPTION
**Pull request by thatguy59**

### CODE REVIEW

File: chapter07/dtj/ex4 [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+print "Tell me about fred and his friends:\n";
+
+my @input = <>;
+
+foreach my $line ( @input ){
+  print $line if( $line =~ /([a-z]+[A-Z]+|[A-Z]+[a-z]+)\s*/);
+
+  # fails if left as because perl automatically assumes $_
+  # print $line if(/([a-z]+[A-Z]+|[A-Z]+[a-z]+)\s*/);
```
Comment (pratyp) [inline chapter07/dtj/ex4:14]: :+1:

File: chapter07/dtj/ex1 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+
+print "Tell me about fred and his friends:\n";
+my @input = <>;
+
+print "\nOutput:\n";
+foreach ( @input ){
```
Comment (glaKris) [inline chapter07/dtj/ex1:10]: foreach( <> )

File: chapter07/dtj/ex4 [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+use warnings;
+use strict;
+
+print "Tell me about fred and his friends:\n";
+
+my @input = <>;
+
+foreach my $line ( @input ){
+  print $line if( $line =~ /([a-z]+[A-Z]+|[A-Z]+[a-z]+)\s*/);
```
Comment (glaKris) [inline chapter07/dtj/ex4:11]: why is a word going to end in a capital? otherwise :+1:

Comment (MarkWignall) [inline chapter07/dtj/ex4:11]: Not quite sure what the +'s and \s\* will help you with as you don't have any bookends to your regex so they should have no effect.

**Pull request merged**